component/prometheus/operator/common/crdmanager.go

```diff
@@ -29,30 +31,43 @@ import (
 // Generous timeout period for configuring all informers
 const informerSyncTimeout = 10 * time.Second
 
-// crdManager is all of the fields required to run the component.
-// on update, this entire thing will be recreated and restarted
+// crdManager is all of the fields required to run a crd based component.
+// on update, this entire thing should be recreated and restarted
 type crdManager struct {
 	mut              sync.Mutex
 	discoveryConfigs map[string]discovery.Configs
 	scrapeConfigs    map[string]*config.ScrapeConfig
-	debugInfo        map[string]*DiscoveredPodMonitor
+	debugInfo        map[string]*operator.DiscoveredResource
 	discoveryManager *discovery.Manager
 	scrapeManager    *scrape.Manager
 
 	opts      component.Options
 	logger    log.Logger
-	args      *Arguments
+	args      *operator.Arguments
 	configGen configgen.ConfigGenerator
+
+	kind string
 }
 
-func NewCRDManager(opts component.Options, logger log.Logger, args *Arguments) *crdManager {
+const (
+	KindPodMonitor     string = "podMonitor"
```

Comment:
we may want to define these constants in component.go (potentially with a type alias) and reference them here to prevent duplication and inconsistencies